### PR TITLE
Allow to break after = for strings

### LINKED ISSUE #1141: array access then assignment
In (with no config):

```js
elements[0].innerHTML = '<div></div><div></div><div></div><div></div><div></div><div></div>';
```

Out:

```js
elements[
  0
].innerHTML = "<div></div><div></div><div></div><div></div><div></div><div></div>";
```

Expected:

```js
elements[0]
  .innerHTML = "<div></div><div></div><div></div><div></div><div></div><div></div>"
```

### PR DESCRIPTION
**Pull request by vjeux**
This is not a general fix but it solves some issues that people are seeing where prettier goes > 80 columns.

Fixes #1141
Also fixes part of #1110

### CODE REVIEW

File: src/printer.js [hunk 1]
```diff
@@ -3094,6 +3094,15 @@ function printAssignment(printedLeft, operator, rightNode, printedRight, options
...
       concat([line, printedRight])
     );
+  } else if (
+    rightNode.type === "StringLiteral" ||
+    (rightNode.type === "Literal" &&
+      typeof rightNode.value === "string")
+  ) {
+    printed = indent(
+      options.tabWidth,
+      concat([line, printedRight])
```
Comment (jlongster) [inline src/printer.js]: Couldn't this be collapsed with the previous `else if`? The contents are the same code.

Comment (vjeux) [inline src/printer.js]: Good eye! I actually copy and pasted this from the one two levels above which does hardline instead of line :)

**Comment by jlongster** [comment]
If you collapse the if statements, you can merge this

**Comment by bakkot** [comment]
Can I ask - why just for strings and binary expressions? It seems a reasonable enough place to break in general.

**Comment by vjeux** [comment]
There are many cases where it looks odd:

```js
var x =
  {
    someObject: true
  };
```

```js
var x =
  someLongCall(
    a
  );
```

**Comment by bakkot** [comment]
Those look fine to me ¯\\_(ツ)_/¯

Better than letting it get longer than 80 characters, anyway.

**Pull request merged**